src/cli/commands/lint.ts

```diff
@@ -178,7 +178,7 @@ const filterResultsBySeverity = (results: IRuleResult[], failSeverity: FailSever
   return results.filter(r => r.severity <= diagnosticSeverity);
 };
 
-const severeEnoughToFail = (results: IRuleResult[], failSeverity: FailSeverity): boolean => {
+export const severeEnoughToFail = (results: IRuleResult[], failSeverity: FailSeverity): boolean => {
```

Comment:
To be honest, I'd rather not export it.
This feels like something that could be easily re-implemented.
Generally speaking, I'm skeptical about exposing everything, as in the end of the day, we make it a part of a public API and therefore need to be more careful when making changes.

Reply:
>could be easily re-implemented

- the mapping between string "error" to numerical "0" would have to be re-implemented
- the filtering logic would have to be re-implemented

Spectral CLI needs this.
Both spectral-action and vscode-spectral would benefit from this.

At the end of the day, it won't be re-implemented. It will be "copy/paste"'d from the source code, and we're the guys maintaining those three products.

>we make it a part of a public API

To be honest, I don't think we really have got a "public API". Everything that needs to be re-used internally or unit-tested is exported.

Maybe should we define a clear boundary. Something like "everything exported from the main entrypoint is public API and won't break during a major version life". 

Everything else is:
 - internal
 - not considered as part of the public API
 - subject to change with any release.

>need to be more careful when making changes.

How about I cover this with unit-tests?

@P0lip  Thoughts?

Reply:
I had expected this sort of thing would pop up after seeing the conversations helping Mike implement Spectral in VS Code. 

If there are common helpers that are used for all three of our own implementations then we're seeing room for improvement for the public API.

![image](https://user-images.githubusercontent.com/67381/80740814-91cb9f80-8b10-11ea-9799-3cd933c5e74d.png)

Reply:
> the mapping between string "error" to numerical "0" would have to be re-implemented

Hm, would it? Each rule will already have an actual DiagnosticSeverity set, so there should be no need

> At the end of the day, it won't be re-implemented. It will be "copy/paste"'d from the source code, and we're the guys maintaining those three products.

Yeah, this is what I meant. The reason I'm against exposing that sort of dangling util from CLI is that we are making a commitment here to support that function.
That being said, if we do want to have a certain amount of of CLI stuff exposed, I suggest we go with something like ESLint does - [CLIEngine](https://eslint.org/docs/developer-guide/nodejs-api#cliengine). This is a bit of a work though, so doesn't need to be done now, just floating an idea.

> To be honest, I don't think we really have got a "public API". Everything that needs to be re-used internally or unit-tested is exported.

Well, I'm fine with exposing this as long as we don't commit to having this given util in this particular spot.
Having said that, if one moves it elsewhere, it's not our fault - they should not be using a function that is not exposed by the main entrypoint.

tl;dr
I'm okay with exporting this util as long as we don't officially recommend using it nor document it. If someone moves it or adjusts the implementation and the consumer breaks, then well, it's not our fault :see_no_evil: